src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/RuntimeLoaderPass.php

```diff
@@ -35,5 +35,10 @@ public function process(ContainerBuilder $container): void
         }
 
         $definition->replaceArgument(0, ServiceLocatorTagPass::register($container, $mapping));
+
+        // Scan all Twig runtimes for attributes
+        if ($container->hasDefinition('twig.extension.attributes')) {
+            $container->getDefinition('twig.extension.attributes')->replaceArgument(0, array_keys($mapping));
```

Comment:
`$mapping` contains too many classes isn't it? We don't need to look for attributes on classes that have a corresponding extension. Can we find a way to make the distinction? Otherwise, we're breaking the declarative native of attributes I think.

Reply:
By using a distinct tag for classes with the attribute `#[AsTwigExtension]` we can be more selective.

Reply:
:rocket: